src/main/java/org/folio/des/service/impl/JobServiceImpl.java

```diff
@@ -187,6 +187,14 @@ public void deleteOldJobs() {
     deleteJobs(jobsToDelete);
   }
 
+  @Transactional
+  @Override
+  public void resend(org.folio.des.domain.dto.Job job) {
```

Comment:
I think we should add the same check as in download() implementation like this:
![image](https://user-images.githubusercontent.com/25097693/191910786-caf7f485-3a76-4401-9b76-ccf7233b6775.png)
Because when files not presented in database - it means they not saved in MINIO and so will not find it in worker module during resend process.
So I think you can introduce new FileNotFoundException and use it in your resend implementation and change download method implementation to throw the same  FileNotFoundException if file is missing

Reply:
Or we can reuse already existed NotFoundException for both download and resend impementation. In this case our REST endpints will return 404 status, because we have Controller Advice for handling NotFoundException.
So lets use this approach

Reply:
done